ppx/melange_ppx.ml

```diff
@@ -697,6 +697,71 @@ module Mapper = struct
             str
         | _ -> super#structure_item str
 
+      method! structure str =
+        let extract_mel_as_ident ~loc x =
+          match x with
+          | PStr [ { pstr_desc = Pstr_eval ({ pexp_desc; _ }, _); _ } ] -> (
+              match pexp_desc with
+              | Pexp_constant (Pconst_string (name, _, _))
+              | Pexp_construct ({ txt = Lident name; _ }, _)
+              | Pexp_ident { txt = Lident name; _ } ->
+                  name
+              | _ ->
+                  Location.raise_errorf ~loc
+                    "Invalid `%@mel.as' payload. Expected a string or an ident."
+              )
+          | _ -> Location.raise_errorf ~loc "Invalid attribute payload."
+        in
+
+        match
+          List.filter_map
+            (function
+              | { pstr_desc = Parsetree.Pstr_value (_, vbs); _ } -> (
+                  match
+                    List.filter_map
+                      (function
+                        | {
+                            pvb_pat = { ppat_desc = Ppat_var pval_name; _ };
+                            pvb_attributes;
+                            pvb_loc;
+                            _;
+                          } -> (
+                            match
+                              Ast_attributes.has_mel_as_payload pvb_attributes
+                            with
+                            | Some ({ attr_payload; _ } as attr) ->
+                                Bs_ast_invariant.mark_used_bs_attribute attr;
```

Comment:
I never get why this is needed, to avoid warnings from ppxlib?

Reply:
Without it, we get warnings:

```
  File "x.ml", line 3, characters 6-12:
  3 | let [@mel.as "put"] other = 2
            ^^^^^^
  Alert unused: Unused attribute [@mel.as]
  This means such annotation is not annotated properly.
  For example, some annotations are only meaningful in externals
```